spec/rubocop/cop/factory_bot/association_style_spec.rb

```diff
@@ -0,0 +1,310 @@
+    end
+
+    context 'when `association` is called with factory option' do
+      it 'registers and corrects an offense' do
+        expect_offense(<<~RUBY)
+          factory :article do
+            association :author, factory: :user
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use implicit style to define associations.
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          factory :article do
+            author factory: %i[user]
+          end
+        RUBY
+      end
+    end
+
+    context 'when `association` is called with array factory option' do
+      it 'registers and corrects an offense' do
+        expect_offense(<<~RUBY)
+          factory :article do
+            association :author, factory: %i[user]
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use implicit style to define associations.
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          factory :article do
+            author factory: %i[user]
+          end
+        RUBY
+      end
+    end
+
+    context 'when `association` is called with trait arguments and factory' \
+            'option' do
+      it 'registers and corrects an offense' do
+        expect_offense(<<~RUBY)
+          factory :article do
+            association :author, :admin, factory: :user
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use implicit style to define associations.
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          factory :article do
+            author factory: %i[user admin]
+          end
+        RUBY
+      end
+    end
+
+    context 'when `association` is called with traits option' do
+      it 'registers and corrects an offense' do
+        expect_offense(<<~RUBY)
+          factory :article do
+            association :author, traits: %i[admin]
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use implicit style to define associations.
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          factory :article do
+            author factory: %i[author admin]
+          end
+        RUBY
+      end
+    end
+
+    context 'when `association` is called with factory and traits options' do
+      it 'registers and corrects an offense' do
+        expect_offense(<<~RUBY)
+          factory :article do
+            association :author, factory: :user, traits: [:admin]
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use implicit style to define associations.
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          factory :article do
+            author factory: %i[user admin]
+          end
+        RUBY
+      end
+    end
+
+    context 'when `association` is called with trait arguments and factory' \
+            'and traits options' do
+      it 'registers and corrects an offense' do
+        expect_offense(<<~RUBY)
+          factory :article do
+            association :author, :active, factory: :user, traits: [:admin]
```

Comment:
`traits:` is not a valid way to pass it, is it?
It even won't be after [your PR](https://github.com/thoughtbot/factory_bot/pull/1552) is merged, as it's only for the implicit syntax?
If we fix this here (and also examples on lines 162 and 179), the `options.delete(:traits)` line and `trait_option_matcher` become redundant.

Reply:
Oh, somehow I had mistakenly thought this had already been implemented. I'll remove the code about `:traits` option 👌